app/uk/gov/hmrc/trusts/controllers/OtherIndividualTransformationController.scala

```diff
@@ -21,49 +21,70 @@ import org.slf4j.LoggerFactory
 import play.api.libs.json.{JsError, JsSuccess, JsValue}
 import play.api.mvc.Action
 import uk.gov.hmrc.trusts.controllers.actions.IdentifierAction
+import uk.gov.hmrc.trusts.models.get_trust_or_estate.get_trust.DisplayTrustNaturalPersonType
 import uk.gov.hmrc.trusts.models.RemoveOtherIndividual
 import uk.gov.hmrc.trusts.models.variation.NaturalPersonType
 import uk.gov.hmrc.trusts.services.OtherIndividualTransformationService
 import uk.gov.hmrc.trusts.utils.ValidationUtil
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class OtherIndividualTransformationController @Inject()(identify: IdentifierAction,
-                                                        transformService: OtherIndividualTransformationService)
-                                                       (implicit val executionContext: ExecutionContext)
-  extends TrustsBaseController with ValidationUtil {
-
+class OtherIndividualTransformationController @Inject()(
+                                          identify: IdentifierAction,
+                                          transformService: OtherIndividualTransformationService
+                                        )(implicit val executionContext: ExecutionContext) extends TrustsBaseController with ValidationUtil {
   private val logger = LoggerFactory.getLogger("application." + this.getClass.getCanonicalName)
 
-  def removeOtherIndividual(utr: String): Action[JsValue] = identify.async(parse.json) {
-    implicit request => {
-      request.body.validate[RemoveOtherIndividual] match {
-        case JsSuccess(otherIndividual, _) =>
-          transformService.removeOtherIndividual(utr, request.identifier, otherIndividual) map { _ =>
-            Ok
-          }
-        case JsError(_) => Future.successful(BadRequest)
+    def removeOtherIndividual(utr: String): Action[JsValue] = identify.async(parse.json) {
+      implicit request => {
+        request.body.validate[RemoveOtherIndividual] match {
+          case JsSuccess(otherIndividual, _) =>
+            transformService.removeOtherIndividual(utr, request.identifier, otherIndividual) map { _ =>
+              Ok
+            }
+          case JsError(_) => Future.successful(BadRequest)
+        }
+      }
+    }
+
+    def amendOtherIndividual(utr: String, index: Int): Action[JsValue] = identify.async(parse.json) {
+      implicit request => {
+        request.body.validate[NaturalPersonType] match {
+          case JsSuccess(otherIndividual, _) =>
+            transformService.amendOtherIndividualTransformer(
+              utr,
+              index,
+              request.identifier,
+              otherIndividual
+            ) map { _ =>
+              Ok
+            }
+          case JsError(errors) =>
+            logger.warn(s"[OtherIndividualTransformationController][addOtherIndividualTransformer] " +
+              s"Supplied json could not be read as an Unidentified Beneficiary - $errors")
```

Comment:
addOtherIndividualTransformer ... should be amendOtherIndividualTransformer

Reply:
The log should also not say unidentified beneficiary

Reply:
Complete